src/Compilers/CSharp/Portable/Parser/LexerCache.cs

```diff
@@ -63,22 +61,23 @@ internal bool TryGetKeywordKind(string key, out SyntaxKind kind)
             return kind != SyntaxKind.None;
         }
 
-        internal SyntaxTrivia LookupTrivia(
+        internal SyntaxTrivia? LookupTrivia(
             char[] textBuffer,
             int keyStart,
             int keyLength,
-            int hashCode,
-            Func<SyntaxTrivia> createTriviaFunction)
+            int hashCode)
```

Comment:
nit: i liked the pattern before.  if we didn't like allocations, could this take a createTriviaFunction, and an arg, allowing the lambda to be static?

Reply:
I like this idea too, it allows the same change to _createQuickTokenFunction (separating apart wouldn't work for that one because of the TextWindow changing when creating the token).